cmd/src/search.go

```diff
@@ -450,17 +497,19 @@ var searchTemplateFuncs = map[string]interface{}{
 		lineNumber := lineMatches[index].(map[string]interface{})["lineNumber"]
 		return prevLineNumber.(float64) == lineNumber.(float64)-1
 	},
-	"searchHighlightMatch": func(match interface{}) string {
+	"searchHighlightMatch": func(content, query, match interface{}) string {
 		m := match.(map[string]interface{})
-		preview := m["preview"].(string)
+		q := query.(string)
 		var highlights []highlight
-		for _, offsetAndLength := range m["offsetAndLengths"].([]interface{}) {
-			ol := offsetAndLength.([]interface{})
-			offset := int(ol[0].(float64))
-			length := int(ol[1].(float64))
-			highlights = append(highlights, highlight{line: 1, character: offset, length: length})
+		if strings.Contains(q, "patterntype:structural") {
```

Comment:
I need a way to branch specifically for this use case. This is my best-current-effort because we don't have a principled/separate package to parse the query. 🌈 RFC 🌈 75 🌈 exists 🌈 to 🌈  help 🌈 with 🌈 this 🌈  problem.

Reply:
This is fine.